ethereum/core/src/test/java/tech/pegasys/pantheon/ethereum/mainnet/precompiles/BLAKE2fPrecompileContractTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.mainnet.precompiles;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+import tech.pegasys.pantheon.ethereum.mainnet.ConstantinopleFixGasCalculator;
+import tech.pegasys.pantheon.ethereum.vm.MessageFrame;
+import tech.pegasys.pantheon.util.bytes.BytesValue;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class BLAKE2fPrecompileContractTest {
+  private final BLAKE2fPrecompileContract contract =
+      new BLAKE2fPrecompileContract(new ConstantinopleFixGasCalculator());
+
+  public BLAKE2fPrecompileContractTest() {}
+
+  private MessageFrame messageFrame = mock(MessageFrame.class);
+
+  @Parameterized.Parameters
+  public static Object[][] parameters() {
+    return new Object[][] {
+      {
+        "6a09e667f2bd8948bb67ae8484caa73b3c6ef372fe94f82ba54ff53a5f1d36f1510e527fade682d19b05688c2b3e6c1f1f83d9abfb41bd6b5be0cd19137e2179000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f202122232425262728292a2b2c2d2e2f303132333435363738393a3b3c3d3e3f0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000c",
```

Comment:
Where do these test cases come from?  Add a link please.  If you create them yourself that is fine, we should fine someone else's test vectors and add them as well, with a link.

Reply:
The test vectors are linked in the PR description above. They come from the go implementation :) Do you mean a comment with the link in the code?

Reply:
Yes, a comment providing a link to the code.  This will be squash merged so patch history doesn't move as well.

Reply:
And for the code impl itself, a link in the javadoc would be good.